pkg/provider/aws/parameterstore/parameterstore.go

```diff
@@ -151,8 +151,30 @@ func (pm *ParameterStore) DeleteSecret(ctx context.Context, remoteRef esv1beta1.
 	return nil
 }
 
-func (pm *ParameterStore) SecretExists(_ context.Context, _ esv1beta1.PushSecretRemoteRef) (bool, error) {
-	return false, errors.New("not implemented")
+func (pm *ParameterStore) SecretExists(ctx context.Context, pushSecretRef esv1beta1.PushSecretRemoteRef) (bool, error) {
+	secretName := pushSecretRef.GetRemoteKey()
```

Comment:
I think this should be `secretName := pm.prefix + data.GetRemoteKey()` since that's what PushSecret is pushing as a secret name.

Reply:
Should be fixed in the latest commit